src/symtable/SymbolTableScope.ts

```diff
@@ -305,17 +310,18 @@ export class SymbolTableScope {
         current block.  The analyze_block() call modifies these
         dictionaries.
         */
-        const tempBound = new Set(bound);
-        const tempFree = new Set(free);
-        const tempGlobal = new Set(global);
+        const tempBound = new Set([...bound]);
```

Comment:
I will be forever amazed by js performance 😄

Reply:
yeah - me too - this one was quite significant maybe 8x slower to do the former.
I guess it's because `Set` is optimized for an Array?
It was actually also quicker to do
```typescript
const tempBound = new Set();
inplaceMerge(tempBond, bound);
```
But only slightly. So I figured this was more readable.

Reply:
🤯